lib/elixir/lib/module/checker.ex

```diff
@@ -0,0 +1,189 @@
+defmodule Module.Checker do
+  def verify(module_map, _binary) do
+    state = %{
+      file: module_map.file,
+      module: module_map.module,
+      compile_opts: module_map.compile_opts,
+      function: nil
+    }
+
+    module_map.definitions
+    |> check_definitions(state)
+    |> List.flatten()
+    |> merge_warnings()
+    |> sort_warnings()
+    |> emit_warnings()
+  end
+
+  defp check_definitions(definitions, state) do
+    Enum.map(definitions, &check_definition(&1, state))
```

Comment:
This should be a `flat_map` so you don't need `List.flatten` above.

Reply:
We need it because we introduce nested lists in other places to avoid repeated `++` operations.